src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedObjects: GroupsMap<T> = {};
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const item of items) {
+    const groupField = String(item[key]);
+
+    if (!groupedObjects[groupField]) {
+      groupedObjects[groupField] = [];
+    }
+
+    groupedObjects[groupField].push(item);
+  }
```

Comment:
Try to use the array method which returns an object for a solution.
So, you do not need object results and can write the method shortly.

Reply:
but why? we use it just here